themes/default/content/docs/guides/testing/unit.md

```diff
@@ -247,7 +247,11 @@ pulumi.runtime.setMocks({
     call: function(args: pulumi.runtime.MockCallArgs) {
         return args.inputs;
     },
-});
+},
+  "project",
+  "stack",
+  false, // Sets the flag `dryRun`, which is true at runtime during a preview.
```

Comment:
I'm not sure how clear this language is. I always find it hard to write comments explaining booleans.
```suggestion
  false, // Sets the flag `dryRun`, which indicates if pulumi is running in preview mode.
```
I'm not sure that my suggestion is better. @cnunciato What do you think about the linguistic clarity of the message.

Reply:
@iwahbe I like your suggestion, yeah -- that seems clearer to me. Although looking at the Node.js SDK docs, should `dryRun` be `preview`?

```typescript
/**
 * setMocks configures the Pulumi runtime to use the given mocks for testing.
 *
 * @param mocks: The mocks to use for calls to provider functions and resource consrtuction.
 * @param project: If provided, the name of the Pulumi project. Defaults to "project".
 * @param stack: If provided, the name of the Pulumi stack. Defaults to "stack".
 * @param preview: If provided, indicates whether or not the program is running a preview. Defaults to false.
 */
export declare function setMocks(mocks: Mocks, project?: string, stack?: string, preview?: boolean): void;
```

Reply:
> So what do y’all think? Any interest in the addition of a short section on this topic, just a couple of sentences to inform readers about the relationship between the `dryRun` flag and previewing, etc?

Yes, this sounds great! The one bit of feedback I have has to do with how this property and the others two (project and stack) are meant to be used in the context of testing. This still isn't clear to me, and I'm sure other users will wonder about it as well, especially when they see them presented alongside `mocks` in this introductory example. (My first question when I saw them was whether they were placeholder values or somehow significant,  and as presented now, it's not clear they're optional.) 

What would you think about pulling all three of these optional properties into this follow-up section as well? That might make the distinction clearer and give you some room to explain how to use them.

Reply:
Yup, on it.

Reply:
> Although looking at the Node.js SDK docs, should `dryRun` be `preview`?

FWIW I do think it should be renamed. But my impression is that since be an API change, it’d be best to propose it in a new issue in pulumi/pulumi — is that right?

> The one bit of feedback I have has to do with how this property and the other two (project and stack) are meant to be used in the context of testing. This still isn't clear to me, and I'm sure other users will wonder about it also, especially when they see them alongside `mocks` in this introductory example. (My first question was whether “project” and “stack” were placeholder values or somehow significant, and as they’re presented now, it's also not clear they're optional.)

Makes 100% sense, this had occurred to me for a fleeting moment but I decided to take a light touch with the changes and see what feedback came back. I’ll address this.

On a related note, fwiw, it seems to me that the name of the method `setMocks` is not fully aligned with how it works / what it does. Yes, you can set the mocks. But it has these additional args that have nothing to do with mocks. They’re more like settings. I understand why they were added; `setMocks` is now used more or less as if it were named `setTestSettings` or something like that. And in fact the docs would be easier to write if it were named something like that. I don’t know if I have the time to make such a proposal to the SDK maintainers, but I figured it couldn’t hurt to share the thought.

Reply:
> FWIW I do think it should be renamed. But my impression is that since be an API change, it’d be best to propose it in a new issue in pulumi/pulumi — is that right?

💯 couldn't agree more! That'd be awesome if you'd be up for it. I will heavily 👍 this proposal. 😄 

> Makes 100% sense, this had occurred to me for a fleeting moment but I decided to take a light touch with the changes and see what feedback came back. I’ll address this.

Awesome, thanks!

> On a related note, fwiw, it seems to me that the name of the method setMocks is not fully aligned with how it works / what it does.

Ah, that's definitely good feedback, yeah -- makes sense given these additional properties.